packages/xdl/src/Hermes.ts

```diff
@@ -0,0 +1,65 @@
+import path from 'path';
+import fs from 'fs-extra';
+
+import spawn from '@expo/spawn-async';
+
+const binPaths: Record<string, string[]> = {
+  win32: ['win64-bin', 'hermes.exe'],
+  linux: ['linux64-bin', 'hermes'],
+  darwin: ['osx-bin', 'hermes'],
+};
+
+async function getBinForPlatform(projectRoot: string): Promise<string | null> {
+  const binPath = binPaths[process.platform];
+  if (!binPath) {
+    throw new Error(`Unsupported platform ${process.platform}`);
+  }
+
+  const expoHermesPath = path.join(
+    projectRoot,
+    'node_modules',
+    '@expo',
+    'hermes-engine',
+    ...binPath
+  );
+  if (await fs.pathExists(expoHermesPath)) {
+    return expoHermesPath;
+  }
+
+  const hermesEnginePath = path.join(projectRoot, 'node_modules', 'hermes-engine', ...binPath);
+  if (await fs.pathExists(hermesEnginePath)) {
+    return hermesEnginePath;
+  }
+  return null;
+}
+
+export async function tryGenerateBytecode(
+  projectRoot: string,
+  jsBundlePath: string,
+  outputPath: string,
+  flags: string[] = []
+) {
+  const hermesFlags = ['-O', ...flags];
+  const binPath = await getBinForPlatform(projectRoot);
+  if (!binPath) {
```

Comment:
Maybe add some warning here?

Reply:
Hermes is not officially supported yet so, for now, it should skip quietly if unavailable